docs/deepops/configuration.md

```diff
@@ -0,0 +1,219 @@
+- `config/group_vars/k8s-cluster.yml`: Variables specific to deploying Kubernetes clusters
+- `config/group_vars/slurm-cluster.yml`: Variables specific to deploying Slurm clusters
+
+It's expected that most DeepOps deployments will make changes to these files!
+The inventory file will be different for every cluster;
+but we also expect that most people will make changes to the Ansible variables that determine how DeepOps behaves.
+While the example configuration contains good defaults (in our opinion!), every cluster is different, and a lot of the features in DeepOps can be configured or are optional.
+
+For example, you may choose to:
+
+- Enable or disable optional features, such as installing a container registry;
+- Change the version of software being installed, such as the Slurm version;
+- Or change the download URL used for a software install, if you want to point to a different mirror
+
+However, unless you're developing changes to DeepOps itself, you should never have to make changes *outside your configuration directory*.
+(If you do, that's a sign that we haven't made our playbooks configurable enough, and you should open an issue or PR to fix that!)
+Ideally, we expect that all customizations for a specific cluster should be made in the configuration directory for that cluster.
+
+
+## Modifying the Ansible inventory
+
+Every cluster deployed with DeepOps will have its own [Ansible inventory file](https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html).
+This file defines the list of hosts that are in your DeepOps cluster, and has several sections depending on how your cluster is configured.
+
+In general, we expect that there will always be a section labeled `[all]` that contains a list of every host in the cluster.
+If you need to supply the IP addresses of your hosts, you will usually specify those IPs here as well.
+For example:
+
+```
+[all]
+my-cluster-controller      ansible_host=10.0.0.1
+my-cluster-compute-01      ansible_host=10.0.0.2
+my-cluster-compute-02      ansible_host=10.0.0.3
+```
+
+The example DeepOps inventory also includes groups for the different components of Kubernetes clusters (`kube-master`, `etcd`, and `kube-node`),
+and groups for the different components of Slurm clusters (`slurm-master` and `slurm-node`).
+These groups are used by DeepOps to determine which playbooks run on which nodes for each type of cluster,
+and you should add nodes to these groups based on how you want to lay out your cluster.
+For example:
+
+```
+[slurm-master]
+my-cluster-controller
+
+[slurm-node]
+my-cluster-compute-01
+my-cluster-compute-02
+```
+
+If you want to further sub-divide your cluster, or target particular hosts with playbooks, you can add other inventory groups to this file!
+For example, if you have a host that you will use as a separate NFS server, you may want to create an `nfs-server` group and only run the [NFS server playbook](../../playbooks/generic/nfs-server.yml) on that host.
```

Comment:
Good example good, should we supply an example kubectl command using the -l flag here? We have them run with -l in a few places in the deployment guide, but I've encountered many users who do not know why they are doing that or realize that it is actually not necessary.